ctru-rs/examples/software-keyboard.rs

```diff
@@ -2,15 +2,38 @@
 //!
 //! This example showcases the use of the Software Keyboard applet to receive text input from the user.
 
-use ctru::applets::swkbd::{Button, SoftwareKeyboard};
+use ctru::applets::swkbd::{Button, CallbackResult, SoftwareKeyboard};
 use ctru::prelude::*;
 
+use std::borrow::Cow;
+use std::ffi::CString;
+
 fn main() {
     let apt = Apt::new().unwrap();
     let mut hid = Hid::new().unwrap();
     let gfx = Gfx::new().unwrap();
     let _console = Console::new(gfx.top_screen.borrow_mut());
 
+    // Prepares a software keyboard with two buttons: one to cancel input and one
+    // to accept it. You can also use `SoftwareKeyboard::new()` to launch the keyboard
+    // with different configurations.
+    let mut keyboard = SoftwareKeyboard::default();
+
+    // Custom filter callback to handle the given input.
+    // Using this callback it's possible to integrate the applet
+    // with custom error messages when the input is incorrect.
+    keyboard.set_filter_callback(Some(Box::new(|str| {
+        // The string is guaranteed to contain valid Unicode text, so we can safely unwrap and use it as a normal `&str`.
```

Comment:
nit: If this is so, we could do this in our library code so the user code doesn't need to assume this invariant. It would make it more Rust-friendly, but I get that it abstracts away the CString, so it's up to you.

Reply:
For the input string, it makes sense to read it as a normal `&str`, since all of its features are known. However, handling the output as a `String` would require reallocations to add the needed `NUL` character at the end.

As we discussed before, the "design problem" comes to the asymmetrical nature of the callback's parameters and return values. I've decided to keep the `&Cstr -> CString` as it more closely relates to what happens behind the scenes.